### PR TITLE
update nikic/php-parser to 4.15.0

### PR DESCRIPTION
**Pull request by staabm**
see https://github.com/nikic/PHP-Parser/releases/tag/v4.15.0

### CODE REVIEW

File: src/Rules/Classes/EnumSanityRule.php [hunk 1]
```diff
@@ -70,15 +69,6 @@ public function processNode(Node $node, Scope $scope): array
 				}
 			}
 
-			// serialize/unserialize is not considered a magic method by PhpParser
```
Comment (staabm) [inline src/Rules/Classes/EnumSanityRule.php:73]: no longer required, as php-parser itself handles this methods as magic.

see changelog
> Added __serialize and __unserialize to magic method list.

**Comment by staabm** [comment]
btw: the new drupal integration test seems to be pretty slow in comparison to other integration test runs. I will have a closer look on that (unrelated to this PR)

**Comment by ondrejmirtes** [comment]
Yes, I want to optimize that. I also need to look into DNF types and implement them in BetterReflection.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**